sdk/Sdk.Generators/FunctionMetadataProviderGenerator/FunctionMetadataProviderGenerator.Parser.cs

```diff
@@ -276,12 +265,12 @@ private bool TryGetParameterInputAndTriggerBindings(MethodDeclarationSyntax meth
                                 hasHttpTrigger = true;
                             }
 
-                            DataType dataType = _dataTypeParser.GetDataType(parameterSymbol.Type);
+                            DataType dataType = _dataTypeParser.GetDataType(parameter.Type);
 
                             bool cardinalityValidated = false;
                             bool supportsDeferredBinding = false;
 
-                            if (SupportsDeferredBinding(attribute, parameterSymbol.Type.ToString()))
+                            if (SupportsDeferredBinding(attribute, parameter.Type.ToString()))
                             {
                                 supportsDeferredBinding = true;
                             }
```

Comment:
Can be `bool supportsDeferredBinding = SupportsDeferredBinding(attribute, parameter.Type.ToString());` ?